qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,192 @@
+        )
+
+    def _create_observable_output(
+        self,
+        ops_ev_mean: np.ndarray,
+        ops_ev_std: np.ndarray,
+        aux_ops: ListOrDict,
+    ) -> ListOrDict[Union[Tuple[np.ndarray, np.ndarray], Tuple[complex, complex]]]:
+        """Creates the right output format for the evaluated auxiliary operators."""
+        operator_number = 0 if aux_ops is None else len(aux_ops)
+        observable_evolution = [(ops_ev_mean[i], ops_ev_std[i]) for i in range(operator_number)]
+
+        if isinstance(aux_ops, dict):
+            observable_evolution = dict(zip(aux_ops.keys(), observable_evolution))
+
+        return observable_evolution
+
+    def _evaluate_aux_ops(
+        self,
+        aux_ops: List[sp.csr_matrix],
+        aux_ops_2: List[sp.csr_matrix],
+        state: np.ndarray,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Evaluates the aux operators if they are provided and stores their value.
+
+        Returns:
+            Tuple of the mean and standard deviation of the aux operators for a given state.
+        """
+        op_mean = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+        op_std = np.sqrt(
+            np.array([np.real(state.conjugate().dot(op2.dot(state))) for op2 in aux_ops_2])
+            - op_mean**2
+        )
+        return op_mean, op_std
+
+    def _start(
+        self, evolution_problem: EvolutionProblem
+    ) -> Tuple[np.ndarray, sp.csr_matrix, sp.csr_matrix, List[sp.csr_matrix], float]:
+        """Returns a tuple with the initial state as an array and the operator needed for time
+            evolution as a sparse matrix.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            A tuple with the initial state as an array, the operator needed for time
+            evolution as a sparse matrix and the number in which to divide the time evolution.
+        """
+
+        # Convert the initial state and hamiltonian into sparse matrices.
+        if isinstance(evolution_problem.initial_state, QuantumCircuit):
+            initial_state = Statevector(evolution_problem.initial_state).data.T
+        else:
+            initial_state = evolution_problem.initial_state.to_matrix(massive=True).transpose()
+
+        hamiltonian = evolution_problem.hamiltonian.to_spmatrix()
+
+        # Get the auxiliary operators as sparse matrices.
+        if isinstance(evolution_problem.aux_operators, list):
+            aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators]
+        elif isinstance(evolution_problem.aux_operators, dict):
+            aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators.values()]
+        else:
+            aux_ops = []
+
+        aux_ops_2 = [aux_op.dot(aux_op) for aux_op in aux_ops]
+
+        timestep = evolution_problem.time / self.timesteps
```

Comment:
Maybe let's restrict this method to operators only and calculate a timestep elsewhere.

Reply:
For Imaginary time evolution that would make sense, but for real time evolution it makes more sense to have it inside of the method. For consistency I believe it should also be inside of the method for Imaginary time evolution.